schematic_api/api/routes.py

```diff
@@ -206,130 +220,183 @@ def get_temp_jsonld(schema_url):
     # get path to temporary JSON-LD file
     return tmp_file.name
 
-# @before_request
-def get_manifest_route(schema_url: str, use_annotations: bool, dataset_ids=None, asset_view = None, output_format=None, title=None, access_token=None):
-    """Get the immediate dependencies that are related to a given source node.
+class ManifestGeneration(BaseModel):
+    schema_url: str
+    data_type: List[str]
+    asset_view: Optional[str] = ''
+    dataset_id: List[str] = []
+    title: Optional[str] = ''
+    output_format: Optional[str] = "google_sheet"
+    use_annotations: Optional[bool] = False
+    
+    def _load_config_(self, app:Flask=app) -> CONFIG:
+        """load configuration file and update asset view if needed
+
         Args:
-            schema_url: link to data model in json ld format
-            title: title of a given manifest. 
-            dataset_id: Synapse ID of the "dataset" entity on Synapse (for a given center/project).
-            output_format: contains three option: "excel", "google_sheet", and "dataframe". if set to "excel", return an excel spreadsheet
-            use_annotations: Whether to use existing annotations during manifest generation
-            asset_view: ID of view listing all project data assets. For example, for Synapse this would be the Synapse ID of the fileview listing all data assets for a given project.
-            access_token: Token
-        Returns:
-            Googlesheet URL (if sheet_url is True), or pandas dataframe (if sheet_url is False).
-    """
+            app (Flask, optional): Flask app object. Defaults to current flask object.
 
-    # call config_handler()
-    config_handler(asset_view = asset_view)
+        Returns:
+            CONFIG: schematic configuration object
+        """
+        return config_handler(app=app, asset_view = self.asset_view)
 
-    # get path to temporary JSON-LD file
-    jsonld = get_temp_jsonld(schema_url)
+    def _check_dataset_match_datatype_(self) -> None:
```

Comment:
Thanks for the comment Tom. I was initially thinking that this function will be internal. But i just made the change to use dataclass for validation instead of using this.